src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/HtmlHelper.cs

```diff
@@ -219,15 +219,10 @@ public virtual void Contextualize([NotNull] ViewContext viewContext)
         }
 
         /// <inheritdoc />
-        public HtmlString AntiForgeryToken()
+        public HtmlString AntiforgeryToken()
         {
-            var tagBuilder = _htmlGenerator.GenerateAntiForgery(ViewContext);
-            if (tagBuilder == null)
-            {
-                return HtmlString.Empty;
-            }
-
-            return tagBuilder.ToHtmlString(TagRenderMode.SelfClosing);
+            var html = _htmlGenerator.GenerateAntiforgery(ViewContext);
+            return html ?? HtmlString.Empty;
```

Comment:
I suspect this is a safeguard against someone implementing `IHtmlGenerator` and returning null.

`DefaultAntiforgery` and `DefaultHtmlGenerator` never return null or empty.